templates/repo/diff/section_unified.tmpl

```diff
@@ -12,21 +12,23 @@
 			{{if eq .GetType 4}}
 				{{if $.root.AfterCommitID}}
 					<td colspan="2" class="lines-num">
-						{{if or (eq $line.GetExpandDirection 3) (eq $line.GetExpandDirection 5)}}
-							<a role="button" class="blob-excerpt" data-url="{{$.root.RepoLink}}/blob_excerpt/{{PathEscape $.root.AfterCommitID}}" data-query="{{$line.GetBlobExcerptQuery}}&style=unified&direction=down&wiki={{$.root.PageIsWiki}}" data-anchor="diff-{{$file.NameHash}}K{{$line.SectionInfo.RightIdx}}">
-								{{svg "octicon-fold-down"}}
-							</a>
-						{{end}}
-						{{if or (eq $line.GetExpandDirection 3) (eq $line.GetExpandDirection 4)}}
-							<a role="button" class="blob-excerpt" data-url="{{$.root.RepoLink}}/blob_excerpt/{{PathEscape $.root.AfterCommitID}}" data-query="{{$line.GetBlobExcerptQuery}}&style=unified&direction=up&wiki={{$.root.PageIsWiki}}" data-anchor="diff-{{$file.NameHash}}K{{$line.SectionInfo.RightIdx}}">
-								{{svg "octicon-fold-up"}}
-							</a>
-						{{end}}
-						{{if eq $line.GetExpandDirection 2}}
-							<a role="button" class="blob-excerpt" data-url="{{$.root.RepoLink}}/blob_excerpt/{{PathEscape $.root.AfterCommitID}}" data-query="{{$line.GetBlobExcerptQuery}}&style=unified&direction=&wiki={{$.root.PageIsWiki}}" data-anchor="diff-{{$file.NameHash}}K{{$line.SectionInfo.RightIdx}}">
-								{{svg "octicon-fold"}}
-							</a>
-						{{end}}
+						<div class="gt-df">
```

Comment:
Wrapper may not be necessary once the mentioned style is fixed.

Reply:
The wrapper is still necessary, otherwise the elements are not centered correctly.

Reply:
Odd, `display: flex` alone should not center.

Reply:
Well, `display: flex` combined with `flex:1` on each element so that each element uses $\frac{1}{n}$ width for $n$ children.
That's what I meant with centered.
Otherwise, the buttons are only as large as necessary, and are at random $(x,y)$ positions.

Reply:
I see. From what I can tell, there can only ever be one button here, but your method would work if there are two or more buttons as I tested it by duplicating the button nodes in the dev tools.

Reply:
Here are two buttons.
![image](https://github.com/go-gitea/gitea/assets/51889757/dbf20b01-6871-48a5-8127-6c24ba3f004e)
That was the the whole problem why I added the wrapper.

Reply:
Ah I see with `$line.GetExpandDirection 3` being true, there could be two buttons, one up, one down. Have never seen this happen every, so I wonder if these `if` conditions even make sense because as far as I know, there can only be three expander variants, up, down and both ways.

Reply:
Yes, these templates need a **huge** rewrite.
They are absolutely not maintainable.
Already the fact that I had to change the same thing in so many goddamn places is not a good sign.

Reply:
Yes, this is one of the ugliest parts of the UI, way too many hacks layered on top of each other.